clients/drcachesim/tracer/raw2trace.cpp

```diff
@@ -1291,15 +1336,61 @@ raw2trace_t::write(void *tls, const trace_entry_t *start, const trace_entry_t *e
                     return error;
                 start = it;
                 DEBUG_ASSERT(tdata->cur_chunk_instr_count == 0);
-                // TODO i#5724: We need to re-emit encodings for "it" and any further
-                // instrs in this buffer: have a callback passed in which constructs
-                // an encoding from an instr record?
             }
-            if (type_is_instr(static_cast<trace_type_t>(it->type)))
+            if (type_is_instr(static_cast<trace_type_t>(it->type))) {
                 ++tdata->cur_chunk_instr_count;
+                ++instr_ordinal;
+                if (instr_ordinal >= static_cast<int>(decode_pcs.size()))
+                    return "decode_pcs is missing entries for written instructions";
```

Comment:
This error-return should probably also be gated by `TESTANY(OFFLINE_FILE_TYPE_ENCODINGS, tdata->file_type)`.

Reply:
Done, plus the size>0 check.